spaghetto/bin/src/Flags.purs

```diff
@@ -4,13 +4,87 @@ import Spago.Prelude
 
 import ArgParse.Basic (ArgParser)
 import ArgParse.Basic as ArgParser
+import Data.Set.NonEmpty (NonEmptySet)
+import Data.Set.NonEmpty as NonEmptySet
+import Node.Path as Path
+import Spago.Core.Config (ShowSourceCode(..))
+import Spago.Core.Config as Core
+import Spago.Paths as Paths
 
 selectedPackage ∷ ArgParser (Maybe String)
 selectedPackage =
   ArgParser.argument [ "--package", "-p" ]
     "Select the local project to build"
     # ArgParser.optional
 
+psaStrict ∷ ArgParser (Maybe Boolean)
+psaStrict =
+  ArgParser.flag [ "--strict" ]
+    "Promotes project sources' warnings to errors"
+    # ArgParser.boolean
+    # ArgParser.optional
+
+psaCensorWarnings ∷ ArgParser (Maybe Boolean)
+psaCensorWarnings =
+  ArgParser.flag [ "--psa-censor-warnings" ]
+    "Censor all warnings"
+    # ArgParser.boolean
+    # ArgParser.optional
+
+psaCensorLib ∷ ArgParser (Maybe Boolean)
+psaCensorLib =
+  ArgParser.flag [ "--psa-censor-lib" ]
+    "Censor warnings from library sources"
+    # ArgParser.boolean
+    # ArgParser.optional
+
+psaCensorSrc ∷ ArgParser (Maybe Boolean)
+psaCensorSrc =
+  ArgParser.flag [ "--psa-censor-src" ]
+    "Censor warnings from project sources"
+    # ArgParser.boolean
+    # ArgParser.optional
+
+psaShowSource ∷ ArgParser (Maybe ShowSourceCode)
+psaShowSource =
+  NoSourceCode
+    <$ ArgParser.flag [ "--psa-no-source" ]
+      "Disable original source code printing"
+    # ArgParser.optional
+
+psaCensorCodes :: ArgParser (Maybe (NonEmptySet String))
+psaCensorCodes =
+  ArgParser.argument [ "--psa-censor-code" ]
+    "Censor a specific error code (e.g. `ShadowedName`)"
+    # ArgParser.many
+    <#> NonEmptySet.fromFoldable
+
+psaFilterCodes :: ArgParser (Maybe (NonEmptySet String))
+psaFilterCodes =
+  ArgParser.argument [ "--psa-filter-code" ]
+    "Only show a specific error code (e.g. `TypesDoNotUnify`)"
+    # ArgParser.many
+    <#> NonEmptySet.fromFoldable
+
+psaStatVerbosity :: ArgParser (Maybe Core.StatVerbosity)
+psaStatVerbosity = ArgParser.optional $ ArgParser.choose "StatVerbosity"
+  [ Core.VerboseStats <$ ArgParser.flag [ "--psa-verbose-stats" ] "Show counts for each warning type"
+  , Core.NoStats <$ ArgParser.flag [ "--psa-censor-stats" ] "Censor warning/error summary"
+  ]
+
+psaStashFile ∷ ArgParser (Maybe (Either Boolean String))
+psaStashFile = ArgParser.optional $ ArgParser.choose "stash"
+  [ ArgParser.argument [ "--psa-stash-file" ] "Enable persistent warnings using a specific stash file"
```

Comment:
The name of this flag is a bit confusing, I'd prefer something more direct (I mean, it doesn't matter if these are persisted to a file or anywhere else) like `--persist-warnings`

Reply:
That can work, but this doesn't just persist warnings. It also persists compiler errors. We could do `--persist`, but then it's not clear what's being saved.